src/Storages/MergeTree/KeyCondition.cpp

```diff
@@ -1486,6 +1486,14 @@ String KeyCondition::RPNElement::toString() const
 
 
 bool KeyCondition::alwaysUnknownOrTrue() const
+{
+    return unknownOrTrue(true);
+}
+bool KeyCondition::mayBeUnknownOrTrue() const
+{
+    return unknownOrTrue(false);
+}
+bool KeyCondition::unknownOrTrue(bool always) const
```

Comment:
It's better to have an `alwaysUnknown` method.

Reply:
>It's better to have an alwaysUnknown method

But ALWAYS_TRUE should negate the optimization

> or else an alwaysTrue condition will negate the optimization.

But returning `true` for `FUNCTION_UNKNOWN` is odd

Will rename them to `indexCannotBeUsed(bool allow_unknown)`